src/main/scala/com/yotpo/metorikku/test/Tester.scala

```diff
@@ -74,162 +74,407 @@ case class Tester(config: TesterConfig) {
-    val selectiveDifferencesActual = columns.map(col => mainDF.select(col).except(subtractDF.select(col)))
-    val diffsArr: Array[String] = selectiveDifferencesActual.filter(d => d.count() > 0).map(diff => dataFrameShowToString(diff))
-    val diffsCount = selectiveDifferencesActual.count(_.count() > 0)
-    if(diffsCount > 0) log.warn(logMessage + "\n" + diffsArr.mkString("\n"))
-    diffsCount
+  private def getConfiguredKeysByTableName(configuredKeys: Option[Map[String, List[String]]], tableName: String) = {
+    configuredKeys match {
+      case Some(tableToKeys) =>
+        tableToKeys.contains(tableName) match {
+          case true => Option(tableToKeys(tableName))
+          case _ => None
+        }
+      case None => None
+    }
   }
 
-  private def getExpectedSchema(expectedSchemaKeys:Iterable[String], actualSchema:StructType): StructType = {
-    var expectedStructFields = Seq[StructField]()
-    for(key <- expectedSchemaKeys){
-      val structFieldMatch = actualSchema.filter(_.name == key)
-      if(structFieldMatch.nonEmpty){
-        val currFieldType = structFieldMatch.head.dataType
-        val fieldType = getExpectedSchemaFieldType(currFieldType)
-        expectedStructFields = expectedStructFields :+ StructField(key, fieldType , true)
-      } else {
-        log.warn(s"The expected schema key : $key doesnt exist in the actual schema, the test will fail because of it")
-      }
+  private def getInvalidConfiguredKeysTable(configuredKeys: List[String],
+                                            allExpectedFields: List[String]): Option[List[String]] = {
+    val invalidKeys = configuredKeys.filter(confKey => !allExpectedFields.contains(confKey))
+    invalidKeys.isEmpty match {
+      case true => None
+      case _ => Option(invalidKeys)
     }
-    StructType(expectedStructFields)
   }
 
-  private def getExpectedArrayFields(expectedSchemaKeys:Iterable[String], actualSchema:StructType): Seq[String] = {
-    var expectedArrayStructFields = Seq[String]()
-    for(key <- expectedSchemaKeys){
-      val currFieldType = actualSchema.filter(_.name == key).head.dataType
-      if(currFieldType.toString.contains("Array")) expectedArrayStructFields = expectedArrayStructFields :+ key
+  private def getInvalidKeysNonExistingTableErrors(allColsKeys: Map[String, List[String]],
+                                                   invalidKeys: List[String],
+                                                   tableName: String) = {
+    val errorData = ErrorMsgData(ErrorType.InvalidKeysNonExisting, tableName, invalidKeys, allColsKeys(tableName))
+    Array(ErrorMsgs.getErrorByType(errorData))
+  }
+
+  def showDfToConsoleOrLogger(redirectDfShowToLogger: Boolean, df: DataFrame, size: Int, truncate: Boolean, lastRowIndex: Int): Unit = {
+    val lastRowIndexStr = lastRowIndex.toString
+    redirectDfShowToLogger match {
```

Comment:
This shouldn't be optional, always print the same way
this function is not needed